app/src/main/java/com/example/vybe/MainActivity.java

```diff
@@ -22,17 +42,88 @@ protected void onCreate(Bundle savedInstanceState) {
+                // User is logged in
+                if (user != null){
+                    Intent intent = new Intent(MainActivity.this, MyVibesActivity.class);
+                    startActivity(intent);
+                }
+            }
+        };
 
         loginButton.setOnClickListener(view -> {
+            // Check that the email and password fields are entered
+            if (!isEmpty(emailField) && !isEmpty(passwordField)){
+                // Get the credentials that were entered
+                String email = getTrimmedString(emailField);
+                String password = getTrimmedString(passwordField);
+                // Sign in using Firebase
+                mAuth.signInWithEmailAndPassword(email, password)
+                        .addOnCompleteListener(new OnCompleteListener<AuthResult>() {
+                    @Override
+                    public void onComplete(@NonNull Task<AuthResult> task) {
+                        // If the task succeed, login passed and go to My Vibes Activity
+                        if (task.isSuccessful()){
+                            Intent intent = new Intent(MainActivity.this, MyVibesActivity.class);
+                            startActivity(intent);
+                        }
+                        else {
+                            Toast.makeText(getApplicationContext(), "Login Failed: " + task.getException().getMessage(), Toast.LENGTH_LONG).show();
+                        }
+                    }
+                });
+            }
 
-            Intent intent = new Intent(MainActivity.this, MyVibesActivity.class);
-            startActivity(intent);
         });
 
         signupButton.setOnClickListener(view -> {
-
-            Intent intent = new Intent(MainActivity.this, MyVibesActivity.class);
+            Intent intent = new Intent(MainActivity.this, CreateAccountActivity.class);
             startActivity(intent);
         });
     }
+
+    // Add the authentication listener on start
+    // This can be used that if the user is already logged in then to redirect to My Vibes Activity
+    @Override
+    protected void onStart() {
+        super.onStart();
+        mAuth.addAuthStateListener(mAuthListener);
+    }
+
+    // Remove the listener on stop as we don't want to redirect? not sure
+    @Override
+    protected void onStop() {
+        super.onStop();
+        if (mAuthListener != null){
+            mAuth.removeAuthStateListener(mAuthListener);
+        }
+    }
+
+    /**
+     * Helper function to check if an edit text field is empty
+     * @param et
+     *      EditText that we want to check
+     * @return Returns true if the field is empty, false it is filled
+     */
+    public static boolean isEmpty(EditText et){
+        return et.getText().toString().trim().length() == 0;
+    }
+
+    /**
+     * Helper function to get trimmed string from edit text field
+     * @param et
+     *      EditText that we want to get the string from
```

Comment:
not 2 be picky but can u just rename this to `edittext`

Reply:
sure thing scrum master